src/bootstrap/builder.rs

```diff
@@ -1543,6 +1543,12 @@ impl<'a> Builder<'a> {
             }
         }
 
+        // For now --generate-link-to-definition is only available on nightly
+        // TODO(SkiFire13): Remove the stage > 0 check when beta reaches 1.56
+        if (cmd == "doc" || cmd == "rustdoc") && stage > 0 {
```

Comment:
I think this is actually unnecessary, sorry to mislead you before. The builder just won't pass the flags unless it's running rustdoc.
```suggestion
        if stage > 0 {
```